### PR TITLE
fix(deploy): reverse order of input error and description

### LINKED ISSUE #2918: Missing/misplaced error messages in Cloud deploy overlay
__Describe the Bug__

The non-empty validation is not appearing for the Client ID and Client secret fields in the Cloud deploy overlay. Additionally, the Audience validation message is incorrectly placed.

__Steps to Reproduce__

1.create a diagram and save
2. open deploy overlay
3. change to Self-managed and click "deploy"

Screenshot 2022-04-12 at 17 43 18

__Expected Behavior__

1. The validation message "... must not be empty." should appear for Client ID and Client secret
2. The validation message for Audience field should appear right bellow the input (before the description)

__Environment__

- Execution Platform: Camunda Cloud

### PR DESCRIPTION
**Pull request by smbea**
Closes #2918

### CODE REVIEW

**Comment by andreasgeier** [comment]
I don't know if this is already the case, so I would like to comment to be on the safe side: I suggest to not fix this on single occasions but make it a general structure of the input component (if such a component exists).

- label
- input
- error message
- description

@philippfromme already fixed this for linting messages in the properties panel so this is the second case of the same problem.

**Comment by smbea** [comment]
> I don't know if this is already the case, so I would like to comment to be on the safe side: I suggest to not fix this on single occasions but make it a general structure of the input component (if such a component exists).
> 
> * label
> * input
> * error message
> * description
> 
> @philippfromme already fixed this for linting messages in the properties panel so this is the second case of the same problem.

I fixed this in the general custom component, meaning it will show for all instances that originate in the Modeler app. But this doesn't apply to cases like the properties panel, since its styles are in a different library.

**Comment by smbea** [comment]
@pinussilvestrus could you take a look at this one?

**Comment by pinussilvestrus** [comment]
Will do, thanks for pinging 👍

File: client/src/shared/ui/form/TextInput.js [hunk 1]
```diff
@@ -95,6 +85,9 @@ export default function TextInput(props) {
         <FormFeedback
           error={ error }
         />
+        {description &&
+          <p className="form-control">{description}</p>
+        }
```
Comment (pinussilvestrus) [inline client/src/shared/ui/form/TextInput.js:90]:
```suggestion
        { description &&
          <p className="form-control">{ description }</p>
        }
```

Comment (pinussilvestrus) [reply client/src/shared/ui/form/TextInput.js:90]: There are still some cases in this file that could need some spaces inside brackets. 🙂

**Review by pinussilvestrus — CHANGES_REQUESTED** [review]
When I press `deploy`, the bug described in the issue still appears:

> The non-empty validation is not appearing for the Client ID and Client secret fields in the Cloud deploy overlay.

image

**Comment by smbea** [comment]
Since this is related to the current saved config, please check if this also works for you @pinussilvestrus

File: client/src/plugins/zeebe-plugin/deployment-plugin/DeploymentPlugin.js [hunk 2]
```diff
@@ -353,24 +353,31 @@ export default class DeploymentPlugin extends PureComponent {
...
       rememberCredentials: false
     };
 
     const previousEndpoints = await this.getEndpoints();
     if (previousEndpoints.length) {
-      endpoint = previousEndpoints[0];
+      endpoint = {
+        ...credentials,
+        ...previousEndpoints[0]
+      };
```
Comment (pinussilvestrus) [inline client/src/plugins/zeebe-plugin/deployment-plugin/DeploymentPlugin.js:380]: Is there any way we can test this, to verify the bug is fixed?

**Review by pinussilvestrus — COMMENTED** [review]
LGTM, good job 👍 
Only one comment regarding test coverage

File: client/src/plugins/zeebe-plugin/deployment-plugin/DeploymentPlugin.js (hunk 2 above)
Comment (smbea) [reply client/src/plugins/zeebe-plugin/deployment-plugin/DeploymentPlugin.js:380]: I will look into adding a test case that ensures this behaviour doesn't happen again

Comment (smbea) [reply client/src/plugins/zeebe-plugin/deployment-plugin/DeploymentPlugin.js:380]: I added a test case to check that the default endpoint has the credential keys, when they haven't been saved to the config

**Review by pinussilvestrus — APPROVED** [review]
LGTM

**Pull request merged**